lib/block_view_validator.go

```diff
@@ -16,27 +16,63 @@ import (
 	"github.com/pkg/errors"
 )
 
-// FIXME: Add a big comment here explaining how validator Register/Unregister/Unjail
-// all fit together at a high level.
+// RegisterAsValidator: Registers a new validator. This transaction can be called multiple times
+// if a validator needs to update any of their registration info such as their domains. Once
+// a validator is registered, stake can be assigned to that validator, the validator is eligible
+// to participate in consensus by voting, and may be selected as leader to propose new blocks.
+//
+// UnregisterAsValidator: Unregisters an existing validator. This unstakes all stake assigned to this
+// validator and removes this validator from the set of eligible validators. A user would have to
+// re-register by submitting a subsequent RegisterAsValidator transaction to be re-included.
+//
+// UnjailValidator: Unjails a jailed validator if sufficient time (epochs) have elapsed since the
+// validator was first jailed. A validator is jailed if they fail to participate in consensus by
+// either voting or proposing blocks for too long. A jailed validator is ineligible to receive
+// any block rewards and ineligible to elected leader.
 
 //
 // TYPES: ValidatorEntry
 //
 
 type ValidatorEntry struct {
+	// The ValidatorPKID is the primary key for a ValidatorEntry. It is the PKID
+	// for the transactor who registered the validator. A user's PKID can ever
+	// only be association with one validator.
 	ValidatorPKID *PKID
+	// Domains is a slice of web domains where the validator can be reached.
 	// Note: if someone is updating their ValidatorEntry, they need to include
 	// all domains. The Domains field is not appended to. It is overwritten.
-	Domains                  [][]byte
-	DisableDelegatedStake    bool
-	VotingPublicKey          *bls.PublicKey
+	Domains [][]byte
+	// DisableDelegatedStake is a boolean that indicates whether the validator
+	// disallows delegated / 3rd party stake being assigned to themselves. If
+	// a validator sets DisableDelegatedStake to true, then they can still
+	// stake with themselves, but all other users will receive an error if they
+	// try to stake with this validator.
+	DisableDelegatedStake bool
+	// The VotingPublicKey is a BLS PublicKey that is used in consensus messages.
+	// A validator signs consensus messages with their VotingPrivateKey and then
+	// other validators can reliably prove the message came from this validator
+	// by verifying against their VotingPublicKey.
+	VotingPublicKey *bls.PublicKey
+	// The VotingPublicKeySiganture is the signature of the SHA256(VotingPublicKey).
```

Comment:
```suggestion
	// The VotingPublicKeySignature is the signature of the SHA256(VotingPublicKey).
```
Also, couldn't some other validator just use the same voting public key and signature since we're only hashing the voting public key? Are we sure we don't want to include the validator's deso public key in the hash?

Reply:
If we just hashed and signed the VotingPublicKey, you would be right. But we hash and sign the TransactorPublicKey not the VotingPublicKey. The VotingPublicKey is used to confirm the signature corresponds to the VotingPrivateKey which signed the SHA256(TransactorPublicKey).

Reply:
So the signature proves both that 1) you own the private key associated with the VotingPublicKey and 2) that this VotingPublicKey is associated with the current TransactorPublicKey.

Reply:
Oh, I see now. I wrote it wrong in this comment. Updating...